packages/react-button/src/components/Button/useButtonClasses.ts

```diff
@@ -1,489 +1,403 @@
-              'var(--button-highContrast-pressed-borderColor, ' +
-              'var(--button-highContrast-hovered-borderColor, ' +
-              'var(--button-highContrast-borderColor)))',
-            color:
-              'var(--button-highContrast-pressed-contentColor, ' +
-              'var(--button-highContrast-hovered-contentColor, ' +
-              'var(--button-highContarst-contentColor)))',
-            [`.${GlobalClassNames.icon}`]: {
-              color:
-                'var(--button-highContrast-pressed-iconColor, ' +
-                'var(--button-highContrast-hovered-iconColor, ' +
-                'var(--button-highContrast-iconColor)))',
-            },
-          },
+const useRootClasses = makeStyles<ButtonStyleSelectors>([
+  [
+    null,
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
+
+      return {
+        ':global(:root)': {
+          // TODO: how to define and access component variables that do not have globals/aliases?
+          //       consider IE11 friendly way
+          '--button-height': '32px',
+          '--button-paddingX': '12px',
+          '--button-minWidth': '96px',
+          '--button-maxWidth': '280px',
+
+          '--button-iconOnly-padding': '6px',
         },
 
-        '&[aria-disabled=true]': {
-          opacity: 'var(--button-disabled-opacity)',
-          backgroundColor: 'var(--button-disabled-background)',
-          color: 'var(--button-disabled-contentColor)',
-          borderColor: 'var(--button-disabled-borderColor)',
-          boxShadow: 'var(--button-disabled-boxShadow)',
-
-          [`.${GlobalClassNames.icon}`]: {
-            color: 'var(--button-disabled-iconColor)',
-          },
-
-          [EdgeChromiumHighContrastSelector]: {
-            background: 'var(--button-highContrast-disabled-background, var(--button-highContrast-background))',
-            borderColor: 'var(--button-highContrast-disabled-borderColor, var(--button-highContrast-borderColor))',
-            color: 'var(--button-highContrast-disabled-contentColor, var(--button-highContrast-contentColor))',
-            opacity: 'var(--button-highContrast-disabled-opacity, var(--button-highContrast-opacity))',
-            [`.${GlobalClassNames.icon}`]: {
-              color: 'var(--button-highContrast-disabled-iconColor, --button-highContrast-iconColor)',
-            },
-          },
-        },
-      },
-    ],
-
-    icon: [
-      GlobalClassNames.icon,
-      {
-        color: 'var(--button-iconColor)',
-        display: 'flex',
-        flexShrink: 0,
+        display: 'inline-flex',
         alignItems: 'center',
         justifyContent: 'center',
-        width: 'var(--button-iconSize)',
-        height: 'var(--button-iconSize)',
-        fontSize: 'var(--button-iconSize, inherit)',
-        fontWeight: 'normal',
-        lineHeight: '1',
-      },
-    ],
-
-    content: {
-      lineHeight: '1',
-    },
-  },
+        gap: buttonTokens.base.iconSpacing,
```

Comment:
FYI: It's fancy property, but is not supported by the most browsers including Safari, https://caniuse.com/?search=gap.

Reply:
Added TODO here for the next iteration. I want to get alpha out to early testing folks. We'll fast follow on this.